contrib/babelfishpg_tds/src/include/tds_request.h

```diff
@@ -506,7 +506,7 @@ SetColMetadataForTvp(ParameterToken temp, const StringInfo message, uint64_t *of
 	char	   *tempString;
 	int			i = 0;
 	char	   *messageData = message->data;
-	StringInfo	tempStringInfo = palloc(sizeof(StringInfoData));
+	StringInfoData	tempStringInfo;
```

Comment:
From what I understand, we needed to `memset` `tempStringInfo.data`.
If yes, is this change from `StringInfo` to `StringInfoData` needed?

Do you also know why setting `data[0] = '\0` was not sufficient?

```/*
 * initStringInfo
 *
 * Initialize a StringInfoData struct (with previously undefined contents)
 * to describe an empty string.
 */
void
initStringInfo(StringInfo str)
{
    int         size = 1024;    /* initial default buffer size */

    str->data = (char *) palloc(size);
    str->maxlen = size;
    resetStringInfo(str);
}

/*
 * resetStringInfo
 *
 * Reset the StringInfo: the data buffer remains valid, but its
 * previous content, if any, is cleared.
 */
void
resetStringInfo(StringInfo str)
{
    str->data[0] = '\0';
    str->len = 0;
    str->cursor = 0;
}
```

Lastly, I suggest we run the failing `TestTvp` with valgrind enabled  to confirm issue is fixed.

Reply:
Technically it is not necessary to change from `StringInfo` to `StringInfoData`, but in other places in the code it's far more common to use StringInfoData in this manner than StringInfo.

I'm not sure exactly why setting data[0] = '\0' is insufficient in this case. 

I ran `TestTvp` with valgrind enabled locally - without this fix, we see the errors as above complaining about uninitialised values; with it, no error.